doc/developer-notes.md

```diff
@@ -513,6 +515,63 @@ To enable test parallelism:
 cmake -DJOBS=$(nproc) -P build/Coverage.cmake
 ```
 
+#### Using LLVM/Clang toolchain
+
+The following generates a coverage report for unit tests and functional tests.
+
+Configure the build with the following flags:
+
+> Consider building with a clean state using `rm -rf build`
```

Comment:
We don't do this in any other docs, and as far as I'm aware, this being needed would be a bug, not sure it should be here.

Reply:
> but if it is important to remove, it could be done in a follow-up.

The more I look at this, the less clear it seems to be. Clearly the dir reuse can/is causing issues, however the discussion in #31942 also claims that any need to nuke a build dir is a bug. If we are going to start adding this, we should at least be consistent, and add this same disclaimer to every bit of documentation that invokes cmake. Will followup in #31942.